src/solver.cpp

```diff
@@ -116,8 +117,15 @@ void Solver::setup(const realtype t0, Model *model, const AmiVector &x0,
 
             applyTolerancesFSA();
         } else if (sensi_meth == SensitivityMethod::adjoint) {
-            /* Allocate space for the adjoint computation */
-            adjInit();
+            if (steadystate) {
+                /* Allocate space for forward quadratures */
+                quadInit(xQ0);
+                applyQuadTolerances();
+                setSparseJacFn_ss();
```

Comment:
Found it easier to overwrite the JacFcn, rather than also changing the intialization function of the linear solver. In the long run, this line should be replaced with a function, with passes the correct Jacobian function depending on the linear solver. So currently, this logic will **only** work with KLU, I'm afraid.

Reply:
if it only works with KLU, please throw an error if a different linear solver is set.